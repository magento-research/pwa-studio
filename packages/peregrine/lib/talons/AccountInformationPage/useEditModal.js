import { useAppContext } from '../../context/app';

export const useEditModal = () => {
    const [{ drawer }, { closeDrawer }] = useAppContext();
    const isOpen = drawer === 'accountInformation.edit';

    return {
        handleClose: closeDrawer,
        isOpen
    };
};
