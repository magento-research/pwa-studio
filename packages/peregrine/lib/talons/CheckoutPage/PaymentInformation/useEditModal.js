import { useState, useCallback } from 'react';

/**
 * Talon to handle checkout page's payment information edit modal.
 *
 * @param {Function} props.onClose callback to be called when the modal's close or cancel button is clicked.
 *
 * @returns {
 *   isLoading: Boolean,
 *   shouldRequestPaymentNonce: Boolean,
 *   handleClose: Function,
 *   handleUpdate: Function,
 *   handlePaymentSuccess: Function,
 *   handleDropinReady: Function,
 *   handlePaymentError: Function
 * }
 */
export const useEditModal = props => {
    const { onClose } = props;

    /**
     * Definitions
     */

    const [isLoading, setIsLoading] = useState(true);
    const [shouldRequestPaymentNonce, setShouldRequestPaymentNonce] = useState(
        false
    );

    /**
     * Helper Functions
     */

    const handleClose = useCallback(() => {
        onClose();
    }, [onClose]);

    const handleUpdate = useCallback(() => {
        setShouldRequestPaymentNonce(true);
    }, [setShouldRequestPaymentNonce]);

    const handlePaymentSuccess = useCallback(() => {
        onClose();
    }, [onClose]);

    const handlePaymentError = useCallback(() => {
        setShouldRequestPaymentNonce(false);
    }, []);

    const handleDropinReady = useCallback(() => {
        setIsLoading(false);
    }, [setIsLoading]);

    return {
        isLoading,
        shouldRequestPaymentNonce,
        handleClose,
        handleUpdate,
        handlePaymentSuccess,
        handlePaymentError,
        handleDropinReady
    };
};
