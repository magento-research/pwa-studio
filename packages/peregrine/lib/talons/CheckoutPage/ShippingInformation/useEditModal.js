import { useAppContext } from '../../../context/app';

export const useEditModal = () => {
    const [{ drawer }, { closeDrawer }] = useAppContext();
    const isOpen = drawer === 'shipping.edit';

    return {
        handleClose: closeDrawer,
        isOpen
    };
};
