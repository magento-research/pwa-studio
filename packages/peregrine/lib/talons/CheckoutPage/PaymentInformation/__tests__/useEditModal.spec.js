import React from 'react';

import createTestInstance from '../../../../util/createTestInstance';
import { useEditModal } from '../useEditModal';

const Component = props => {
    const talonProps = useEditModal(props);

    return <i talonProps={talonProps} />;
};

const getTalonProps = props => {
    const tree = createTestInstance(<Component {...props} />);
    const { root } = tree;
    const { talonProps } = root.findByType('i').props;

    const update = newProps => {
        tree.update(<Component {...newProps} />);

        return talonProps;
    };

    return { talonProps, tree, update };
};

test('Snapshot test', () => {
    const { talonProps } = getTalonProps({ onClose: () => {} });

    expect(talonProps).toMatchSnapshot();
});

test('Should call onClose when handleClose is called', () => {
    const onClose = jest.fn();

    const { talonProps } = getTalonProps({ onClose });
    const { handleClose } = talonProps;

    handleClose();

    expect(onClose).toHaveBeenCalled();
});

test('Should call onClose when handlePaymentSuccess is called', () => {
    const onClose = jest.fn();

    const { talonProps } = getTalonProps({ onClose });
    const { handlePaymentSuccess } = talonProps;

    handlePaymentSuccess();

    expect(onClose).toHaveBeenCalled();
});

test('Should set shouldRequestPaymentNonce to true when handleUpdate is called', () => {
    const { talonProps, tree } = getTalonProps({
        onClose: () => {}
    });

    talonProps.handleUpdate();

    expect(
        tree.root.findByType('i').props.talonProps.shouldRequestPaymentNonce
    ).toBeTruthy();
});

test('Should set isLoading to false when handleDropinReady is called', () => {
    const { talonProps, tree } = getTalonProps({
        onClose: () => {}
    });

    talonProps.handleDropinReady();

    expect(tree.root.findByType('i').props.talonProps.isLoading).toBeFalsy();
});
