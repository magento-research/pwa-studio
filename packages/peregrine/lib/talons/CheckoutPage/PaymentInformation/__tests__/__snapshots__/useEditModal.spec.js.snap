// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
Object {
  "handleClose": [Function],
  "handleDropinReady": [Function],
  "handlePaymentSuccess": [Function],
  "handleUpdate": [Function],
  "isLoading": true,
  "shouldRequestPaymentNonce": false,
}
`;
